spec/models/manageiq/providers/hawkular/middleware_manager/event_catcher_spec.rb

```diff
@@ -0,0 +1,50 @@
+describe ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcher do
+  before do
+    allow(MiqServer).to receive(:my_zone).and_return("default")
+    auth = AuthToken.new(:name => "test", :auth_key => "valid-token", :userid => "jdoe", :password => "password")
+    @ems = FactoryGirl.create(:ems_hawkular,
+                              :hostname        => 'localhost',
+                              :port            => 8080,
+                              :authentications => [auth])
+
+    WebMock.allow_net_connect!
+  end
+
+  include ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcherMixin
```

Comment:
I think what you actually want to test is the `MiddlewareEventMonitor`. This releases you from this heavy setup and makes the test much cleaner. It also makes the inclusion of the `EventCatcherMixin` go away.
Feel free to ping me on gitter. I'm in CET too and happy to pair up :)

https://github.com/ManageIQ/manageiq/blob/099656e8698c55fcd30b1851da7d950f96f39bed/gems/manageiq-providers-amazon/spec/models/manageiq/providers/amazon/cloud_manager/event_catcher/stream_spec.rb